### ISSUE TITLE
Taker can take offer with same id twice

### ISSUE THREAD

**Issue by da-kami** [itchysats/itchysats#324]
Not sure how I managed to do this, but it seems the maker's offer is not always removed correctly. 

Saw this in the taker logs, which means that we tried to take the same offer twice:

```
2021-10-14 20:24:47 ERROR Message handler failed: error returned from database: UNIQUE constraint failed: cfds.order_uuid: UNIQUE constraint failed: cfds.order_uuid
2021-10-14 20:24:58  INFO POST /api/calculate/margin text/plain:  
```

The maker should not allow this.

**Comment by da-kami** [comment]
This is related to https://github.com/comit-network/hermes/issues/323 - because we don't remove/replace the offer upon auto-reject upon retry the taker will try to create another cfd with the same order. 
Note: Yes, the taker creates a cfd before sending the take request to the maker to keep track of the state.

**Comment by bonomat** [comment]
Imho this has also todo with the db issue: #326.

**Comment by luckysori** [comment]
I can still reproduce this on master (4ffbebda189320d48fd53399aa0a61d7b30bcbbe). It occurred after attempting to send and order request to the maker whilst the maker hadn't yet replied to my original order request.

**Comment by bonomat** [comment]
@da-kami : this is assigned to you and in progress . If you are not working on this, please unasign yourself.

**Comment by scratchscratchscratchy** [comment]
fixed with #532
